src/cmap/conn/command.rs

```diff
@@ -168,53 +171,33 @@ impl RawCommandResponse {
             .map_err(|_| Error::invalid_authentication_response(mechanism_name))
     }
 
-    /// Deserialize the raw bytes into a response backed by a `Document` for further processing.
-    pub(crate) fn into_document_response(self) -> Result<DocumentCommandResponse> {
-        let response: CommandResponse<Document> = self.body()?;
-        Ok(DocumentCommandResponse { response })
+    pub(crate) fn as_is_master_response(&self, round_trip_time: Duration) -> Result<IsMasterReply> {
```

Comment:
since this performs a somewhat costly conversion, this would be better named as `to_is_master_response`. Check out the [API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv) for detailed info on when to use `to` / `as` / `into`.